lib/commands/run/helper.js

```diff
@@ -0,0 +1,94 @@
+const fs = require('fs');
+const ResourcesConfig = require('@src/model/resources-config');
+const CONSTANTS = require('@src/utils/constants');
+const stringUtils = require('@src/utils/string-utils');
+const CliError = require('@src/exceptions/cli-error');
+const NodejsRunFlow = require('./run-flow/nodejs-run');
+const PythonRunFlow = require('./run-flow/python-run');
+const JavaRunFlow = require('./run-flow/java-run');
+
+const RUN_FLOWS = [
+    JavaRunFlow,
+    NodejsRunFlow,
+    PythonRunFlow
+];
+
+module.exports = {
+    getHostedSkillInvocationInfo,
+    getNonHostedSkillInvocationInfo,
+    getSkillCodeFolderName,
+    getNormalisedRuntime,
+    selectRunFlowClass,
+    getSkillFlowInstance
+};
+
+function getHostedSkillInvocationInfo(runtime) {
+    return CONSTANTS.HOSTED_SKILL_INVOCATION_CONFIG.get(runtime);
+}
+
+function getNonHostedSkillInvocationInfo(runtime, handler, skillCodeFolderName) {
+    console.log(runtime);
+    if (!stringUtils.isNonBlankString(runtime)) {
+        throw new CliError('Missing runtime info in '
+            + `resource file ${CONSTANTS.FILE_PATH.ASK_RESOURCES_JSON_CONFIG}.`);
+    }
+    if (!stringUtils.isNonBlankString(handler)) {
+        throw new CliError('Missing handler info '
+            + `in resource file ${CONSTANTS.FILE_PATH.ASK_RESOURCES_JSON_CONFIG}.`);
+    }
+    if (runtime === CONSTANTS.RUNTIME.JAVA) {
+        return { handlerName: handler, skillCodeFolderName };
+    }
+    const handlerInfo = handler.split('.');
+    const handlerName = handlerInfo.pop();
+    const skillFileName = handlerInfo.pop();
+    return { handlerName, skillFileName, skillCodeFolderName };
+}
+
+function getSkillCodeFolderName(profile, skillCodeRegion) {
+    const userRegionSkillCodeFolderName = ResourcesConfig.getInstance().getCodeSrcByRegion(profile, skillCodeRegion);
+    let skillCodeFolderName = ((!stringUtils.isNonBlankString(userRegionSkillCodeFolderName))
+        && (userRegionSkillCodeFolderName !== CONSTANTS.ALEXA.REGION.DEFAULT))
+        ? ResourcesConfig.getInstance().getCodeSrcByRegion(profile, CONSTANTS.ALEXA.REGION.DEFAULT) : userRegionSkillCodeFolderName;
+    if (!stringUtils.isNonBlankString(skillCodeFolderName)) {
+        throw new CliError(`Invalid code setting in region ${skillCodeRegion}. "src" must be set if you want to run `
+            + 'the skill code with skill package.');
+    } else {
+        skillCodeFolderName = ResourcesConfig.getInstance().getCodeSrcByRegion(profile, CONSTANTS.ALEXA.REGION.DEFAULT);
+    }
```

Comment:
do you need this else? seems the value will be set using default region already from line 50